### ISSUE TITLE
Remove deprecated backwardsCompatImplements/fixClassImplements

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5034]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5034|
|Type|task|
|Created|2011-04-07 16:33:22Z|
|Branch|https://github.com/twisted/twisted/tree/impl-components-5034|

[twisted/python/components.py](twisted/python/components.py) has the following note (added in r17451):

```python
## backwardsCompatImplements and fixClassImplements should probably stick around for another
## release cycle. No harm doing so in any case.
```

And there's a reference to non-existing Twisted 2.3:

```python
warnings.warn("components.backwardsCompatImplements doesn't do anything in Twisted 2.3, stop calling it.", ComponentsDeprecationWarning, stacklevel=2)
```

Let's remove this.

<details><summary>Searchable metadata</summary>

```
trac-id__5034 5034
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_impl_components_5034 branches/impl-components-5034
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1302194002000000 1302194002000000
changetime__1331141703000000 1331141703000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [33524]) Branching to 'impl-components-5034'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [33525]) remove deprecated t.p.components methods, introduced in r16264. refs [#5034](https://github.com/twisted/twisted/issues/5034)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for working on this, thijs. [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/impl-components-5034) look good.

There are some lingering references to the removed code:

```
doc/core/upgrades/2.0/components.xhtml:base="twisted.python.components">backwardsCompatImplements</code> to fix this.
doc/core/upgrades/2.0/components.xhtml:twisted.python.components.backwardsCompatImplements(C)
doc/core/upgrades/2.0/components.xhtml:base="twisted.python.components">fixClassImplements</code>: </p>
doc/core/upgrades/2.0/components.xhtml:twisted.python.components.fixClassImplements(o.__class__)
```

Surely everything in `doc/core/upgrades` is no longer relevant to anyone and we can remove it, either here or in a new ticket that gets resolved prior to this one.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [33561]) address review comments, refs [#5034](https://github.com/twisted/twisted/issues/5034)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [jesstess](#note_5):
> Surely everything in `doc/core/upgrades` is no longer relevant to anyone and we can remove it, either here or in a new ticket that gets resolved prior to this one.

Thanks for the review. I'll remove the references in components.xhtml and opened [#5484](https://github.com/twisted/twisted/issues/5484) for the removal of `doc/core/upgrades`.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [33562]) address review comments, refs [#5034](https://github.com/twisted/twisted/issues/5034)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst set owner to @thijstriemstra|
|-|-|

Thanks for working on cleaning up Twisted! Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33659]) Merge impl-components-5034: Remove backwardsCompatImplements and fixClassImplements from twisted.python.components, deprecated in 2006.

Author: thijs
Reviewer: jesstess, itamar
Fixes: [#5034](https://github.com/twisted/twisted/issues/5034)
